lib/GIL/BasicBlock.cpp

```diff
@@ -53,28 +52,16 @@ void BasicBlock::removeInstruction(InstBase *inst)
     _instructions.remove(inst);
 }
 
-InstBase *BasicBlock::getTerminator()
-{
-    if (_instructions.empty() /*|| !_instructions.back().isTerminator()*/)
-        return nullptr;
-    return &_instructions.back();
-}
-
 TerminatorInst *BasicBlock::getTerminatorInst()
 {
-    InstBase *terminator = getTerminator();
-    if (terminator) {
-        return static_cast<TerminatorInst *>(terminator);
-    }
-    return nullptr;
+    if (_instructions.empty())
+        return nullptr;
+    return llvm::dyn_cast<TerminatorInst>(&_instructions.back());
 }
```

Comment:
no use of the `isTerminator()` function ?

Reply:
isTerminator does a llvm::isa (which returns a boolean), llvm::dyn_cast returns a pointer or null. why do the check twice?